### PR TITLE
chore: replaced Logger with backstage implementation LoggerService

### PR DESCRIPTION
**Pull request by 04kash**
- Replaced ```Logger``` with backstage implementation ```LoggerService```
-  Renamed `plugins/feedback-backend/dynamic/inedx.ts` to `plugins/feedback-backend/dynamic/index.ts`

Closes issue #1623

### CODE REVIEW

File: plugins/keycloak-backend/src/providers/KeycloakOrgEntityProvider.ts [hunk 1]
```diff
@@ -269,7 +269,7 @@ export class KeycloakOrgEntityProvider implements EntityProvider {
 }
```
Comment (Zaperex) [inline plugins/keycloak-backend/src/providers/KeycloakOrgEntityProvider.ts:269]: You missed the `Logger` usage in https://github.com/janus-idp/backstage-plugins/blob/835e63f3995fb840108118681f6e012f7ac88f08/plugins/keycloak-backend/src/providers/KeycloakOrgEntityProvider.ts#L183

File: plugins/ocm-backend/dev/types.ts [hunk 2]
```diff
@@ -10,10 +10,10 @@ import { Config } from '@backstage/config';
 import { IdentityApi } from '@backstage/plugin-auth-node';
 import { PermissionEvaluator } from '@backstage/plugin-permission-common';
 
-import { Logger } from 'winston';
+import { LoggerService } from '@backstage/backend-plugin-api';
```
Comment (Zaperex) [inline plugins/ocm-backend/dev/types.ts:14]: For the OCM backend plugin, you need to do the following modifications:
Remove this import:
https://github.com/janus-idp/backstage-plugins/blob/835e63f3995fb840108118681f6e012f7ac88f08/plugins/ocm-backend/src/providers/module.ts#L17
Change remove the wrapper and just pass `logger,`:
https://github.com/janus-idp/backstage-plugins/blob/835e63f3995fb840108118681f6e012f7ac88f08/plugins/ocm-backend/src/providers/module.ts#L40
Change all `winston.Logger` instances to `LoggerService` in https://github.com/janus-idp/backstage-plugins/blob/835e63f3995fb840108118681f6e012f7ac88f08/plugins/ocm-backend/src/providers/ManagedClusterProvider.ts

File: plugins/rbac-backend/src/service/router.ts [hunk 3]
```diff
@@ -2,10 +2,10 @@ import { errorHandler } from '@backstage/backend-common';
 
 import express from 'express';
 import Router from 'express-promise-router';
-import { Logger } from 'winston';
+import { LoggerService } from '@backstage/backend-plugin-api';
```
Comment (Zaperex) [inline plugins/rbac-backend/src/service/router.ts:6]: Similarly for the rbac-backend, the `loggerToWinstonLogger` wrappers should be removed:
https://github.com/janus-idp/backstage-plugins/blob/835e63f3995fb840108118681f6e012f7ac88f08/plugins/rbac-backend/src/plugin.ts#L1

**Review by Zaperex — CHANGES_REQUESTED** [review]
@04kash can you remove the `winston` dependencies in the `package.json` for the packages that no longer use the winston logger? You will also need to update the `packages/cli/templates/default-backend-plugin/package.json.hbs` for new plugins.

Also I noticed you did not remove the `loggerToWinstonLogger` wrappers in the plugins. I commented about the ocm-backend and rbac-backend, but for the rest, I'll just list here:
- ocm-backend
- rbac-backend
- packages/cli backend plugin template 
- 3scale-backend
- aap-backend
- keycloak-backend
- notifications-backend
- orchestrator-backend

**Comment by batzionb** [comment]
verified orchestrator plugin

File: plugins/ocm-backend/dev/types.ts (hunk 2 above)
Comment (04kash) [reply plugins/ocm-backend/dev/types.ts]: @Zaperex I've made these changes

**Comment by Zaperex** [comment]
Please resolve the merge conflicts

**Review by Zaperex — APPROVED** [review]
/lgtm

File: plugins/kiali-backend/src/plugin.ts [hunk 4]
```diff
@@ -22,7 +21,7 @@ export const kialiPlugin = createBackendPlugin({
         catalogApi: catalogServiceRef,
       },
       async init({ http, logger, config }) {
-        const winstonLogger = loggerToWinstonLogger(logger);
+        const winstonLogger = logger;
         const router = await createRouter({
           logger: winstonLogger,
```
Comment (schultzp2020) [inline plugins/kiali-backend/src/plugin.ts:26]:
```suggestion
        const router = await createRouter({
          logger,
```

File: plugins/notifications-backend/src/plugin.ts [hunk 5]
```diff
@@ -49,7 +48,7 @@ export const notificationBackendPlugin = createBackendPlugin({
         identity,
         tokenManager,
       }) {
-        const winstonLogger = loggerToWinstonLogger(logger);
+        const winstonLogger = logger;
         const router = await createRouter({
           logger: winstonLogger,
```
Comment (schultzp2020) [inline plugins/notifications-backend/src/plugin.ts:53]:
```suggestion
        const router = await createRouter({
          logger,
```

File: plugins/rbac-backend/src/plugin.ts [hunk 6]
```diff
@@ -54,7 +53,7 @@ export const rbacPlugin = createBackendPlugin({
         auth,
         httpAuth,
       }) {
-        const winstonLogger = loggerToWinstonLogger(logger);
+        const winstonLogger = logger;
```
Comment (schultzp2020) [inline plugins/rbac-backend/src/plugin.ts:57]:
```suggestion
```

Sadly it's not allowing me to make the full suggestion. Do the same thing as above and remove `winstonLogger`.

File: plugins/rbac-backend/src/role-manager/role-manager.ts [hunk 7]
```diff
@@ -15,7 +14,7 @@ export class BackstageRoleManager implements RoleManager {
   private maxDepth?: number;
   constructor(
     private readonly catalogApi: CatalogApi,
-    private readonly log: Logger,
+    private readonly log: LoggerService,
```
Comment (schultzp2020) [inline plugins/rbac-backend/src/role-manager/role-manager.ts:17]:
```suggestion
    private readonly logger: LoggerService,
```
We should change this to be uniform with other implementations.

**Review by schultzp2020 — CHANGES_REQUESTED** [review]

File: plugins/keycloak-backend/src/providers/KeycloakOrgEntityProvider.ts [hunk 8]
```diff
@@ -30,7 +31,7 @@ import KcAdminClient from '@keycloak/keycloak-admin-client';
 import type { Credentials } from '@keycloak/keycloak-admin-client/lib/utils/auth';
 import { merge } from 'lodash';
 import * as uuid from 'uuid';
-import { Logger } from 'winston';
+
```
Comment (schultzp2020) [inline plugins/keycloak-backend/src/providers/KeycloakOrgEntityProvider.ts:34]:
```suggestion
```

File: plugins/rbac-backend/src/file-permissions/csv-file-watcher.test.ts [hunk 9]
```diff
@@ -684,7 +685,7 @@ describe('CSVFileWatcher', () => {
 async function createEnforcer(
   theModel: Model,
   adapter: Adapter,
-  log: Logger,
+  log: LoggerService,
```
Comment (schultzp2020) [inline plugins/rbac-backend/src/file-permissions/csv-file-watcher.test.ts:688]:
```suggestion
  logger: LoggerService,
```

File: plugins/rbac-backend/src/role-manager/ancestor-search-memo.ts [hunk 10]
```diff
@@ -62,7 +61,7 @@ export class AncestorSearchMemo {
     return this.graph.hasNode(groupRef);
   }
 
-  debugNodesAndEdges(log: Logger, userEntity: string): void {
+  debugNodesAndEdges(log: LoggerService, userEntity: string): void {
```
Comment (schultzp2020) [inline plugins/rbac-backend/src/role-manager/ancestor-search-memo.ts:64]:
```suggestion
  debugNodesAndEdges(logger: LoggerService, userEntity: string): void {
```

File: plugins/3scale-backend/src/module.ts [hunk 11]
```diff
@@ -36,7 +35,7 @@ export const catalogModule3ScaleEntityProvider = createBackendModule({
       async init({ catalog, config, logger, scheduler }) {
         catalog.addEntityProvider(
           ThreeScaleApiEntityProvider.fromConfig(config, {
-            logger: loggerToWinstonLogger(logger),
+            logger: logger,
```
Comment (schultzp2020) [inline plugins/3scale-backend/src/module.ts:38]:
```suggestion
            logger,
```
Please change every instance of longhand to shorthand.

File: plugins/aap-backend/src/module.ts [hunk 12]
```diff
@@ -36,7 +35,7 @@ export const catalogModuleAapResourceEntityProvider = createBackendModule({
       async init({ catalog, config, logger, scheduler }) {
         catalog.addEntityProvider(
           AapResourceEntityProvider.fromConfig(config, {
-            logger: loggerToWinstonLogger(logger),
+            logger: logger,
```
Comment (schultzp2020) [inline plugins/aap-backend/src/module.ts:38]:
```suggestion
            logger,
```

File: plugins/kiali-backend/src/clients/fetch.ts [hunk 13]
```diff
@@ -32,11 +33,11 @@ export type KialiValidations = {
...
 export class KialiFetcher {
-  private readonly logger: Logger;
+  private readonly logger: LoggerService;
   private kialiAuth: KialiAuthentication;
   private KialiDetails: KialiDetails;
 
-  constructor(KD: KialiDetails, log: Logger) {
+  constructor(KD: KialiDetails, log: LoggerService) {
     this.KialiDetails = KD;
     this.logger = log;
```
[the remark this answers is not in the document]
Comment (schultzp2020) [reply plugins/kiali-backend/src/clients/fetch.ts:42]: Please change all instances of applicable `log` to `logger`

**Review by schultzp2020 — CHANGES_REQUESTED** [review]

File: plugins/scaffolder-annotator-action/src/actions/annotator/mocks/catalog-info.yaml [hunk 14]
```diff
@@ -1,13 +1,15 @@
 apiVersion: backstage.io/v1alpha1
```
Comment (schultzp2020) [inline plugins/scaffolder-annotator-action/src/actions/annotator/mocks/catalog-info.yaml:1]: I think you accidently edited this file.

**Review by schultzp2020 — CHANGES_REQUESTED** [review]

**Review by schultzp2020 — APPROVED** [review]
/approve
/lgtm

**Comment by Zaperex** [comment]
@04kash can you rerun `yarn install`, you have a dirty yarn lock right now.

**Comment by schultzp2020** [comment]
@04kash there are typescript errors that need to be resolved.

File: plugins/3scale-backend/src/providers/ThreeScaleApiEntityProvider.ts [hunk 15]
```diff
@@ -170,7 +170,7 @@ export class ThreeScaleApiEntityProvider implements EntityProvider {
           service.service.id,
         );
         if (apiDoc !== undefined) {
-          this.logger.info(apiDoc);
+          this.logger.info(`${apiDoc}`);
```
Comment (schultzp2020) [inline plugins/3scale-backend/src/providers/ThreeScaleApiEntityProvider.ts:173]:
```suggestion
          this.logger.info(JSON.stringify(apiDoc));
```

File: plugins/orchestrator-backend/src/service/ScaffolderService.ts [hunk 16]
```diff
@@ -78,7 +78,7 @@ export class ScaffolderService {
         workingDirectory,
         actionExecutionContext.instanceId ?? randomUUID(),
       );
-    } catch (err) {
+    } catch (err: any) {
```
Comment (schultzp2020) [inline plugins/orchestrator-backend/src/service/ScaffolderService.ts:81]:
```suggestion
    } catch (err: unknown) {
```

File: plugins/orchestrator-backend/src/service/ScaffolderService.ts [hunk 17]
```diff
@@ -78,7 +78,7 @@ export class ScaffolderService {
         workingDirectory,
         actionExecutionContext.instanceId ?? randomUUID(),
       );
-    } catch (err) {
+    } catch (err: unknown) {
       this.logger.error(
         `Error getting working directory to execute action ${actionExecutionContext.actionId}`,
         err,
```
Comment (Zaperex) [inline plugins/orchestrator-backend/src/service/ScaffolderService.ts:84]:
```suggestion
        err as JSONObject,
```

**Review by Zaperex — APPROVED** [review]
/lgtm

**Pull request merged**